src/chains/ethereum/ethereum/src/blockchain.ts

```diff
@@ -596,7 +596,7 @@ export default class Blockchain extends Emittery<BlockchainTypedEvents> {
     await this.#blockBeingSavedPromise;
     return {
       transactions,
-      blockNumber: nextBlock.header.number.toBuffer()
+      blockNumber: nextBlock.header.number.toArrayLike(Buffer)
```

Comment:
We already polyfill Buffer in the browser, but BN doesn't handle it well, so we have to do it this way .